src/key-handler/index.tsx

```diff
@@ -61,11 +67,11 @@ export function KeyHandler(props: KeyHandlerProps) {
         );
       };
     },
-    [focusedStack, focusGroupId, handler, triggers]
+    [focusedStack, focusGroupId, rest, triggers]
```

Comment:
Yeah, this is the part that is a little awkward. Passing in `rest` as a dependency means it gets re-triggered on every render since `rest` is a new object every time we call this function. We need some way of getting something where the identity is the same on every render. Maybe something like this:

```ts
const handler = "handler" in rest ? rest.handler : null;
```

And then in the dependency array:

```suggestion
    [focusedStack, focusGroupId, handler, triggers]
```

Reply:
@scotttrinh  but would handler we have passed to the array change if the actual handler (in rest) gets changed?

Reply:
yeah, `handler` in this case is a reference to `rest.handler` so if that value changes, `handler` also changes, but if it stays the same, it's referentially equal (`===`).

Reply:
To be clear, we'd want to use `handler` within the effect, too, not `rest.handler`.